net/instaweb/rewriter/public/rewrite_options.h

```diff
@@ -1043,6 +1044,32 @@ class RewriteOptions {
     set_option(level, &level_);
   }
 
+  // returns true iff given name and value are valid to set as a http header.
+  // If false is returned, error_message will have a descriptive failure
+  // message set.
+  bool ValidateConfiguredHttpHeader(const GoogleString& name,
+                                    const GoogleString& value,
+                                    GoogleString* error_message);
+
+  // If false was returned, no changes are made, and error_message will be
+  // assigned a failure description. If true was returned, the name/value
+  // passed validation and have been appended to resource_headers_
+  bool ValidateAndAddResourceHeader(const StringPiece& name,
+                                    const StringPiece& value,
+                                    GoogleString* error_message);
+
+  // Unconditionally appends the given name / value to resource_headers_. Use
+  // ValidateAndAddResourceHeader if you need validation.
```

Comment:
If I have header a:b and then add header a:c, we'll wind up with both, right?  That's fine; but please doc it.